### PR TITLE
Fix array variance checks for portable span

### PR DESCRIPTION
**Pull request by shiftylogic**
Fix for #14533

### CODE REVIEW

**Comment by ahsonkhan** [comment]
LGTM

**Comment by ahsonkhan** [comment]
To clarify, do we want to keep the type mismatch check for Span?

https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/Span.cs#L33

cc @jkotas?

**Comment by shiftylogic** [comment]
Span needs to stay as it is, based on my conversation with @jkotas. In the array case, if you assign a string array to an object array and attempt to write anything but a string to it, checks are in-place to throw. However, those same checks don't exist for Span. So, a writable span without this check would be incorrect.

I'll let @jkotas jump in if I misspoke.

File: src/System.Memory/src/System/ReadOnlySpan.cs [hunk 1]
```diff
@@ -29,8 +29,6 @@ public ReadOnlySpan(T[] array)
         {
             if (array == null)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
-            if (default(T) == null && array.GetType() != typeof(T[]))
```
Comment (jkotas) [inline src/System.Memory/src/System/ReadOnlySpan.cs]: There are multiple constructor that take array. It should be removed in all of them. Note that this code is slow span specific - your tests should catch it if you run them on slow span.

Comment (shiftylogic) [inline src/System.Memory/src/System/ReadOnlySpan.cs]: I added netfx to the test configuration to force the tests to run against slow span.

Note that my change relies on #17388 being merged, so this PR cannot merge before #17388.

Comment (shiftylogic) [inline src/System.Memory/src/System/ReadOnlySpan.cs]: I cleaned up my environment and reverted the cherrypicked PR I thought I needed. It appears I may not need that change.

**Comment by ahsonkhan** [comment]
Part of https://github.com/dotnet/corefxlab/issues/1314

**Pull request merged**